### PR TITLE
RDISCROWD-1500 Bypass two-factor authentication via config list

### PR DESCRIPTION
**Pull request by primaryobjects**
- Added config option to bypass two-factor authentication based upon login email address.
- Support for automated testing specific login email address.

```python
# Enable two factor authentication
ENABLE_TWO_FACTOR_AUTH = True
BYPASS_TWO_FACTOR_AUTH = ['user1@user.com', 'user2@user.com']
```

### CODE REVIEW

File: pybossa/otp.py [hunk 1]
```diff
@@ -73,3 +73,10 @@ def retrieve_email_for_token(token):
 def expire_token(token):
     key = _create_url_token_key(token)
     conn.delete(key)
+
+def is_enabled(user_email, config):
+    # Allow bypassing two-factor-auth if the feature is disabled or if the user's email is included in the bypass list.
+    bypass_list = getattr(config, 'BYPASS_TWO_FACTOR_AUTH', [])
```
Comment (fiorda) [inline pybossa/otp.py]: You should be able to do `config.get('BYPASS_TWO_FACTOR_AUTH', [])`

File: test/test_otp.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
+import json
 from default import with_context
 from pybossa import otp
 
+class JSONObject:
```
Comment (fiorda) [inline test/test_otp.py]: why do you need this?

File: pybossa/otp.py (hunk 1 above)
Comment (primaryobjects) [reply pybossa/otp.py]: Thank you. Yes, I realized this well from the web unit test failing. Working on an update. :)

File: pybossa/otp.py [hunk 3]
```diff
@@ -73,3 +73,10 @@ def retrieve_email_for_token(token):
 def expire_token(token):
     key = _create_url_token_key(token)
     conn.delete(key)
+
+def is_enabled(user_email, config):
+    # Allow bypassing two-factor-auth if the feature is disabled or if the user's email is included in the bypass list.
+    bypass_list = config.get('BYPASS_TWO_FACTOR_AUTH', [])
+    enable_two_factor_auth = config.get('ENABLE_TWO_FACTOR_AUTH', False)
+
+    return enable_two_factor_auth and (bypass_list is None or not user_email in bypass_list)
```
Comment (fiorda) [inline pybossa/otp.py]: This is good - fyi you can also do `... or user_email not in bypass_list` - also if you don't want to check if `bypass_list` is `None` you can do
`bypass_list = config.get('BYPASS_TWO_FACTOR_AUTH') or []`

Comment (primaryobjects) [reply pybossa/otp.py]: Thank you. That was the syntax I was looking for. :)
https://github.com/bloomberg/pybossa/pull/269/commits/a6892133c8398e36fe0bd4518ae8d1b5dccf2c6a

**Pull request merged**